Source/Core/VideoCommon/TextureInfo.cpp

```diff
@@ -0,0 +1,325 @@
+// Copyright 2021 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "VideoCommon/TextureInfo.h"
+
+#include <xxhash.h>
+
+#include "Common/Align.h"
+#include "Core/HW/Memmap.h"
+#include "VideoCommon/BPMemory.h"
+#include "VideoCommon/SamplerCommon.h"
+#include "VideoCommon/TextureDecoder.h"
+
+TextureInfo::TextureInfo(u32 stage)
```

Comment:
I'm not sure I like how this constructor pulls a whole bunch of info out of the emulated state. Seems wrong that something called `TextureInfo` has a constructor that is anything other than pure. 

Instead, I recommend having a static method called something like `TextureInfo::FromCurrentState(u32 stage)` that returns a constructed TextureInfo.